Tests/Particles/ManagedCUDADeposition/make.out

```diff
@@ -0,0 +1,14 @@
+Loading ../../..//Tools/GNUMake/comps/pgi.mak...
```

Comment:
I think I have a slight preference for the current approach, which makes `AMReX_Particles.H` uglier but easier to understand IMO, since you'd have to go to a different file to see that CudaManagedAllocator may or may not do something with related to Cuda. But, I can change it if you prefer.